codec/test_codec.go

```diff
@@ -887,3 +947,53 @@ func TestMultipleTags(codec GeneralCodec, t testing.TB) {
 		require.Empty(output.NoTags)
 	}
 }
+
+func TestMap(codec GeneralCodec, t testing.TB) {
```

Comment:
Do we have a test where the key is an interface?

Reply:
Added in de80f0d54